packages/cli/generators/openapi/templates/src/models/model-template.ts.ejs

```diff
@@ -0,0 +1,28 @@
+import {model, property} from '@loopback/repository';
+
+<%_
+const _referencedModels = locals.referencedModels || [];
+const _properties = locals.properties || [];
+_referencedModels.forEach(m => {
+%>
+import {<%- m.className %>} from './<%- m.fileName.substring(0, m.fileName.length -3) %>';
+<%_
+});
+-%>
+/**
+ * The model class is generated from OpenAPI schema - <%- name %>
+ */
+@model({name: '<%- name %>'})
```

Comment:
Hmm, is it a good idea to represent all OAS definitions as models? I am expecting design-first APIs to have multiple fine-grained schemas that don't map 1:1 to persistence models. Consider for example LB 3.x, where we have `Product` and `Product_new` schemas to represent payload of update vs. create requests (the later must omit properties like `id` and `_rev`).

IMO, we should represent schemas as interfaces.

If you still think schemas should be models, then please:
 - Add a constructor argument accepting a subset of properties to allow easy initialization (`constructor(data?: Partial<<%- className %>>)`).
 - Consider inheriting from `Model` or `Entity` base class. My main concern is whether our type system allows a free-form class not inheriting from Model/Entity to be used with a Repository?

Reply:
We need more discussion for the schema to LB4 mapping.

1. LB4 models are just schema-backed classes for data objects. They can be used for persistence as well as data transfer. 

2. `class` captures more information from the schema, such as `default` and validation constraints.

3. For `anyOf`, `oneOf`, `allOf` based composition, I map them to TS type aliases for union and intersection types.

4. I'm open to generate both classes and interfaces. A model class can implement the interface as the base. We can even expose the option via CLI. We can implement it as follow-up.

5. I added the `constructor(data?: Partial<MyModel>` in the generated model class.

Reply:
Sounds good.